### PR TITLE
Expand the set of units supported by the prometheus exporter

### PR DESCRIPTION
**Pull request by dashpole**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/4372.

We currently have a very small set of units we recognize.  Expand it so that we can cover more cases, such as the one pointed out in the issue.

Change the tests to use seconds instead of milliseconds, as we want to ensure that case works correctly.

The set of units is copied from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/prometheus/normalize_name.go#L19

### CODE REVIEW

File: exporters/prometheus/exporter.go [hunk 1]
```diff
@@ -330,9 +330,45 @@ func sanitizeRune(r rune) rune {
...
+	"TiBy": "_tibibytes",
+	"KBy":  "_kilobytes",
+	"MBy":  "_megabytes",
+	"GBy":  "_gigabytes",
+	"TBy":  "_terabytes",
+	"B":    "_bytes",
+	"KB":   "_kilobytes",
+	"MB":   "_megabytes",
+	"GB":   "_gigabytes",
+	"TB":   "_terabytes",
```
Comment (MrAlias) [inline exporters/prometheus/exporter.go:356]: "B" is a bel in the UCUM: https://ucum.org/ucum#section-Alphabetic-Index-By-Symbol

```suggestion
```

File: exporters/prometheus/exporter.go [hunk 2]
```diff
@@ -330,9 +330,45 @@ func sanitizeRune(r rune) rune {
...
+	"J": "_joules",
+	"W": "_watts",
+	"g": "_grams",
+
+	// Misc
+	"Cel": "_celsius",
+	"Hz":  "_hertz",
+	"1":   "_ratio",
+	"%":   "_percent",
+	"$":   "_dollars",
```
Comment (MrAlias) [inline exporters/prometheus/exporter.go:371]: This isn't a valid UCUM unit.

```suggestion
```

File: exporters/prometheus/exporter.go (hunk 1 above)
Comment (dashpole) [reply exporters/prometheus/exporter.go]: removed

File: exporters/prometheus/exporter.go (hunk 2 above)
Comment (MrAlias) [reply exporters/prometheus/exporter.go:366]: This doesn't look resolved. Is there a missing commit that needs to be pushed?

Comment (dashpole) [reply exporters/prometheus/exporter.go]: pushed

**Review by MrAlias — APPROVED** [review]

**Comment by pellared** [comment]
Opportunistic question: Should we add some doc comments to https://github.com/open-telemetry/opentelemetry-go/blob/main/unit/unit.go#L17 that we recommend using UCUM units?

**Comment by pellared** [comment]
Opportunistic question: Should we add some doc comments to https://pkg.go.dev/go.opentelemetry.io/otel/metric#WithUnit that we recommend using UCUM units?

**Review by pellared — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Pull request merged**